elastic/kibana
# Fleet Privileges Display

## Summary

Fixed privileges display for features/subFeatures that require all spaces.

### Before
Role privileges display for only `Default` space selected

<img width="728" alt="Screenshot 2024-12-17 at 13 32 17" src="https://github.com/user-attachments/assets/151b7012-aa1a-430c-be22-cc91e64362e3" />

Privileges summary display for only `Default` space selected

<img width="471" alt="Screenshot 2024-12-17 at 13 32 50" src="https://github.com/user-attachments/assets/964c2223-163d-4081-a37d-196f5df5df5c" />

### After
Role privileges display for only `Default` space selected

<img width="739" alt="Screenshot 2024-12-17 at 13 30 00" src="https://github.com/user-attachments/assets/0f98a9d7-211d-46ec-82c6-25d29a44be6b" />

Privileges summary display for only `Default` space selected

<img width="569" alt="Screenshot 2024-12-17 at 13 30 19" src="https://github.com/user-attachments/assets/932771fd-6486-4b7e-9de5-6cd34ab74dc9" />

### How to test
With `Default` space:
1. Navigate to Creating a new Role and assign Kibana privileges.
2. Set the Spaces to `Default` Space and the privilege level to All.
3. Navigate to Management category and verify that Fleet is set to `None`.
4. Click on "View privilege summary" and verify that Fleet is set to `None`.

With `*All Spaces`:
1. Navigate to Creating a new Role and assign Kibana privileges.
2. Set the Spaces to `*All Spaces` and the privilege level to All.
3. Navigate to Management category and verify that Fleet is set to `All`
4. Click on "View privilege summary" and verify that Fleet is set to `All`


### Checklist

Check the PR satisfies following conditions. 


__Fixes: https://github.com/elastic/kibana/issues/194686__

## Release Note
Fixed privileges display for features/subFeatures that require all spaces.
